### ISSUE TITLE
Modelchoice segfault with mac

### ISSUE THREAD

**Issue by fradav** [diyabc/abcranger#34]
```text
Francois-Davids-iMac:data fradav$ ../../build/abcranger -n 1000 -t 5
> Model Choice <
///////////////////////////////////////// read headers
read headers done.
///////////////////////////////////////// read reftable
read reftable done.
///////////////////////////////////////// First forest (training on ABC output)
Growing trees ..
 [==================================================] (5/5)   35.1 it/sec
Computing prediction error ..
abcranger(597,0x10bd5a5c0) malloc: Incorrect checksum for freed object 0x7fb01a50ed80: probably modified after being freed.
Corrupt value: 0x66
abcranger(597,0x10bd5a5c0) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6
```

**Comment by fradav** [comment]
Ok pinned to the `free/destroy` policy of apple libc++, probably more restrictive than others. Adjusted.

**Comment by fradav** [comment]
with the help of clang sanitizer, fixed in d8509062b9a0bc124f99a6710f14d2ec0097902d
